app/presenters/ht_registration_presenter.rb

```diff
@@ -79,11 +83,17 @@ def show_detail_display_name
   end
 
   def show_detail_edu_person_principal_name
-    env["HTTP_X_SHIB_EDUPERSONPRINCIPALNAME"]
+    name = env["HTTP_X_SHIB_EDUPERSONPRINCIPALNAME"] || ""
+    # Check if principal name or email matches the given DSP email
+    badge = name == applicant_email ? :ok : :mismatch
```

Comment:
I don't know if we necessarily want to convey "danger" here -- they aren't necessarily expected to match, but it's worth checking that it at least makes sense. It's also not unexpected for this header to be missing (not all identity providers provide it)